services/AppVeyor.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const BaseService = require('./BaseService');
+const loadLogos = require('../lib/load-logos');
+
+/**
+ * AppVeyor CI integration.
+ */
+module.exports = class AppVeyor extends BaseService {
+  async handle({repo, branch}) {
+    let apiUrl = 'https://ci.appveyor.com/api/projects/' + repo;
+    if (branch != null) {
+      apiUrl += '/branch/' + branch;
+    }
+    const {buffer, res} = await this._sendAndCacheRequest(apiUrl, {
+      headers: { 'Accept': 'application/json' }
+    });
+
+    if (res.statusCode === 404) {
+      return {text: 'project not found or access denied'};
+    }
+
+    const data = JSON.parse(buffer);
+    const status = data.build.status;
+    if (status === 'success') {
+      return {text: 'passing', colorscheme: 'brightgreen'};
+    } else if (status !== 'running' && status !== 'queued') {
+      return {text: 'failing', colorscheme: 'red'};
+    } else {
+      return {text: status};
+    }
+  }
+
+  // Metadata
+  static get category() {
+    return 'build';
+  }
+
+  static get uri() {
+    return {
+      format: '/appveyor/ci/([^/]+/[^/]+)(?:/(.+))?',
+      capture: ['repo', 'branch']
+    };
+  }
+
+  static get defaultBadgeData() {
+    return {
+      logo: loadLogos().appveyor,
```

Comment:
I'd rather handle this with a string that gets resolved to the real logo at the time of SVG rendering. It makes testing easier since you can just examine the string instead of the object. Can certainly address that in a follow-up.

Reply:
Good idea!